warehouse/query-core/src/main/java/datawave/query/tables/keyword/KeywordQueryLogic.java

```diff
@@ -154,14 +154,11 @@ public GenericQueryConfiguration initialize(final AccumuloClient client, final Q
             end = PARENT_ONLY;
         }
 
-        // Extract the list of individual documents we're looking for.
-        final Collection<String> queryTerms = extractQueryTerms(settings);
-
-        // Configure the documentLanguageMap based on the data included for each document
-        final Map<String,String> documentLanguageMap = createDocumentLanguageMap(queryTerms);
-        config.getState().getDocumentLanguageMap().putAll(documentLanguageMap);
+        // Populate the identifier and language maps based on the data included for each document in the query.
+        extractIdentifiersAndLanguages(settings, config.getState().getIdentifierMap(), config.getState().getLanguageMap());
 
         // Configure ranges for finding content.
+        final Collection<String> queryTerms = extractQueryTerms(settings);
```

Comment:
these are being extracted in the method above, maybe add this above then pass the result into extractIdentifiersAndLanguages?